### PR TITLE
Rm old solidity

### PR DESCRIPTION
**Pull request by RensR**
This PR removes the old 0.4 and 0.5 Solidity contracts. Some of them were used in 0.8 Hardhat tests, so some migrations have been done from 0.4 to 0.8 for a few testhelpers. 

One test now fails, which is disabled for now until the operator forwarder team can address this test `cannot call functions on the LINK token through callbacks`

### CODE REVIEW

**Review by RyanRHall — COMMENTED** [review]
all automation stuff looks fine to me 👍

File: contracts/test/v0.8/operatorforwarder/Operator.test.ts [hunk 1]
```diff
@@ -1204,15 +1204,15 @@ describe('Operator', () => {
...
-      it('cannot call functions on the LINK token through callbacks', async () => {
-        await evmRevert(
-          maliciousRequester.request(
-            specId,
-            link.address,
-            ethers.utils.toUtf8Bytes('transfer(address,uint256)'),
-          ),
-        )
-      })
+      // it('cannot call functions on the LINK token through callbacks', async () => {
```
Comment (patrickhuie19) [inline contracts/test/v0.8/operatorforwarder/Operator.test.ts:1207]: /nit add a TODO with ticket to add this back in or remove? 

i.e. 
```
// TODO: BCI-9999
```

Comment (RensR) [reply contracts/test/v0.8/operatorforwarder/Operator.test.ts:1207]: There should the ticket live? There are no actual owners

**Review by patrickhuie19 — APPROVED** [review]

File: contracts/test/v0.8/operatorforwarder/Operator.test.ts (hunk 1 above)
Comment (patrickhuie19) [reply contracts/test/v0.8/operatorforwarder/Operator.test.ts:1207]: You could put something in foundations if you wanted

**Pull request merged**